typing/typeclass.ml

```diff
@@ -1620,26 +1637,38 @@ let final_decl env define_class
       in
       raise(Error(cl.pci_loc, env, Unbound_type_var(printer, reason)))
   end;
-
-  (id, cl.pci_name, clty, ty_id, cltydef, obj_id, obj_abbr, cl_id, cl_abbr,
-   arity, pub_meths, coe, expr,
-   { ci_loc = cl.pci_loc;
-     ci_virt = cl.pci_virt;
-     ci_params = ci_params;
-(* TODO : check that we have the correct use of identifiers *)
-     ci_id_name = cl.pci_name;
-     ci_id_class = id;
-     ci_id_class_type = ty_id;
-     ci_id_object = obj_id;
-     ci_id_typehash = cl_id;
-     ci_expr = expr;
-     ci_decl = clty;
-     ci_type_decl = cltydef;
-     ci_attributes = cl.pci_attributes;
- })
+  {
+    clsres_id = id;
```

Comment:
As per https://github.com/ocaml/ocaml/issues/7927#issuecomment-474041486, it would be good in this refactoring to be making use of punning wherever possible - so arguments like this could be renamed to `clsres_id` and then the "transforming" (e.g. from `cl.pci_name` to `clsres_id_loc`) becomes more visually obvious.